KifuSF/CreateDonationViewController.swift

```diff
@@ -0,0 +1,246 @@
+        }
+
+        contentScrollView.updateBottomPadding(keyboardHeight + 20)
+    }
+
+    //TODO: this method gets called twice find out why
+    @objc private  func keyboardWillHide(_ notification: Notification) {
+        contentScrollView.updateBottomPadding(pickUpAddressButton.frame.height + 24)
+    }
+
+    @objc private func dismissVC() {
+        dismiss(animated: true)
+    }
+
+    @objc private func dismissKeyboard() {
+        view.endEditing(true)
+    }
+
+    @objc private func imageViewTapped() {
+        imageHelper.presentActionSheet(from: self)
+    }
+
+    @objc private func pickUpAddressButtonTapped() {
+        guard let _ = userSelectedAProfileImage,
+            let image = descriptorView.imageView.image,
+            let title = titleInputView.contentView.textField.text, !title.isEmpty,
+            let description = descriptionInputView.contentView.text, description != CreateDonationViewController.descriptionPlaceholder else {
+                return showErrorMessage("Please complete all the fields")
+        }
+
+        let locationPicker = LocationPickerViewController()
+
+        locationPicker.showCurrentLocationInitially = true
+        locationPicker.searchBarPlaceholder = "Choose Pickup location"
+        locationPicker.mapType = .standard
+
+        locationPicker.showCurrentLocationButton = true
+
+        locationPicker.completion = { location in
+            guard let location = location else {
+                return print("no location selected")
+            }
+
+            self.pickupLocation = location
+        }
+
+        self.navigationController?.pushViewController(locationPicker, animated: true)
+    }
+
+    private func showErrorMessage(_ errorMessage: String) {
+        errorLabel.isHidden = false
+        errorLabel.text = errorMessage
+
+        UIView.animate(withDuration: UIView.microInteractionDuration, animations: { [unowned self] in
+            self.view.layoutIfNeeded()
+        }, completion: { [unowned self] (_) in
+            self.contentScrollView.scrollToBottom()
+        })
+
+        pickUpAddressButton.resetState()
+    }
+}
+
+//MARK: - UITextFieldDelegate
+extension CreateDonationViewController: UITextFieldDelegate {
+    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
+        descriptionInputView.contentView.becomeFirstResponder()
+
+        return true
+    }
+}
+
+//MARK: - UITextViewDelegate
+extension CreateDonationViewController: UITextViewDelegate {
+    func textViewDidBeginEditing(_ textView: UITextView) {
+        if textView.textColor == .kfPlaceholderText {
+            textView.textColor = .black
+            textView.text = nil
+        }
+    }
+
+    func textViewDidEndEditing(_ textView: UITextView) {
+        if textView.text.isEmpty {
+            textView.textColor = .kfPlaceholderText
+            textView.text = CreateDonationViewController.descriptionPlaceholder
+        }
+    }
```

Comment:
This placeholder text for the `UITextView` could be abstracted away from the ViewController and into a subclass of `UITextView`. **Butt**, since it's working now, i'd leave it as it is now 👍

Reply:
Let's leave this for a future PR

Reply:
I agree